### DISCUSSION TITLE
Add more inline editable tables

### DISCUSSION THREAD

**Discussion by tvdeyen** [solidusio/solidus#4620] [New Features or Ideas]
As a follow up to the `inline-editable-table` introduced in #1580 we should have a look into all tables in the admin and see where this new UI component could fit. Also we need to update the style guide

- [ ] Update style guide and show the new UI component
- [ ] Compile a list with tables in the admin that could make use of this component

**Comment by Mandily** [comment]
@mtomov worked on a PR that added inline editing capabilities to the image variants screen to make it quicker and easier for admins to update fields. I think this could be extended to other tables in the admin so that users could both edit and create simple table rows inline instead of needing to go to other pages. I've broken up the opportunities into:
1. Simply making existing inline editing capabilities available upon hovering of rows. Sometimes we'd be able to remove the edit button from the side if we were also removing a separate edit page, but in the case of pages with a large number of inputs, we'd keep the edit buttons to be able to link the user to that page.
2. Making interactions consistent and moving _save_ or _update_ functions tied to a table row into icons located at the right side of the table, and removing them from underneath the table.
3. Moving _edit_ functionality into a table from a separate page (and thus removing the need for the separate page and for an edit button at the right side of the table)
4. Moving _new_ functionality into a table from a separate page (and thus removing the need for a separate page and for a new button in the header or elsewhere on the page.)

I’ve made a list of tables and what would need to be done to move edit and new functionality into the table [here](https://docs.google.com/spreadsheets/d/13N_inWDx-JCAu6orKDNZbKquL_xMoA6Ky9YH48a9H8Q/edit?usp=sharing)

Let me know if there's any questions/concerns about this.
